### PR TITLE
Restore/proof tests

### PR DESCRIPTION
**Pull request by Patrik-Stas**

### CODE REVIEW

File: libvcx/src/v3/handlers/proof_presentation/prover/prover.rs [hunk 1]
```diff
@@ -34,6 +35,9 @@ impl Prover {
     pub fn retrieve_credentials(&self) -> VcxResult<String> {
         trace!("Prover::retrieve_credentials >>>");
         let presentation_request = self.prover_sm.presentation_request().request_presentations_attach.content()?;
+        if settings::indy_mocks_enabled() {
+            return Ok(CREDS_FROM_PROOF_REQ.to_string());
```
Comment (mirgee) [inline libvcx/src/v3/handlers/proof_presentation/prover/prover.rs:39]: Which tests do require this?

File: libvcx/src/disclosed_proof.rs [hunk 2]
```diff
@@ -1024,33 +1036,40 @@ mod tests {
...
-        assert_eq!(VcxStateType::VcxStateRequestReceived as u32, get_state(handle).unwrap());
+        let handle_proof = create_proof("TEST_CREDENTIAL", &request).unwrap();
+        assert_eq!(VcxStateType::VcxStateRequestReceived as u32, get_state(handle_proof).unwrap());
 
-        send_proof(handle, connection_h).unwrap();
-        assert_eq!(VcxStateType::VcxStateAccepted as u32, get_state(handle).unwrap());
+        // todo: mock dependencies to make generate_proof work
+        // generate_proof(proof_handle, selected_credentials.into(), "{}".to_string()).unwrap();
+        // send_proof(handle_proof, connection_h).unwrap();
+        // assert_eq!(VcxStateType::VcxStateAccepted as u32, get_state(handle_proof).unwrap());
```
Comment (mirgee) [inline libvcx/src/disclosed_proof.rs:1072]: Should we leave this test here? We removed another test requiring credentials...

File: libvcx/src/disclosed_proof.rs [hunk 3]
```diff
@@ -1141,12 +1162,12 @@ mod tests {
     #[cfg(feature = "general_test")]
     fn test_deserialize_succeeds_with_self_attest_allowed() {
         let _setup = SetupDefaults::init();
+        ::settings::set_config_value(::settings::CONFIG_PROTOCOL_TYPE, "4.0");
 
-        let handle = create_proof("id", ::utils::constants::PROOF_REQUEST_JSON).unwrap();
+        let handle_1 = create_proof("id", ARIES_PROOF_REQUEST_PRESENTATION).unwrap();
```
Comment (mirgee) [inline libvcx/src/disclosed_proof.rs:1167]: handle

File: libvcx/src/proof.rs [hunk 4]
```diff
@@ -1061,50 +1022,46 @@ pub mod tests {
...
-        assert_eq!(proof.state(), VcxStateType::VcxStateOfferSent as u32);
-
-        proof.update_state_with_message(ARIES_PROOF_PRESENTATION).unwrap();
-
-        assert_eq!(proof.state(), VcxStateType::VcxStateAccepted as u32);
+        settings::set_config_value(settings::CONFIG_PROTOCOL_TYPE, "4.0");
+
+        let handle_conn = build_test_connection_inviter_requested();
+
+        let hande_proof = create_proof("1".to_string(),
```
Comment (mirgee) [inline libvcx/src/proof.rs:1056]: handle

File: libvcx/src/v3/handlers/proof_presentation/prover/prover.rs (hunk 1 above)
Comment (Patrik-Stas) [reply libvcx/src/v3/handlers/proof_presentation/prover/prover.rs]: Using bit of a new approach here, I am not mocking this explicitly via `set_mock_creds_retrieved_for_proof_request`

File: libvcx/src/lib.rs [hunk 5]
```diff
@@ -48,7 +48,7 @@ pub mod error;
 pub mod credential;
 pub mod object_cache;
 pub mod disclosed_proof;
-
+pub mod mock_settings;
```
Comment (mirgee) [inline libvcx/src/lib.rs:51]: Please move to utils

**Pull request merged**